### DISCUSSION TITLE
StabilityDays and Kustomize

### DISCUSSION THREAD

**Discussion by knechtionscoding** [renovatebot/renovate#18443] [Request Help] [answered]
In the documentation I see:

```
If this is set to a non-zero value, and an update has a release timestamp header, then Renovate will check if the "stability days" have passed.
```

1. How can I know if an update has a release timestamp header?
1. Does that mean that if I set stabilityDays on something that doesn't have a release timestamp header renovate will hold the PR forever?

I'm mostly trying to ensure that my kustomize updates are held for a certain period of time to ensure nothing is broken in them, but also I don't want to hold them forever.

**Comment by rarkins** [comment, marked as answer]
1. It depends if the datasource of the dependency supports timestamps (it's not dependent on the manger, like kustomize)
2. The setting will be ignored if no release timestamp is present

**Reply by knechtionscoding** [reply]
So how can I verify the support on the source of the dependency then?
Cool. Good to know.

**Reply by rarkins** [reply]
I'm not sure there's any summary, but if a datasource includes `releaseTimestamp` in its logic then it does. https://github.com/renovatebot/renovate/search?q=releasetimestamp

The thing is not every *registry* might include it, e.g. maybe Nexus does not, even if npmjs does

**Reply by knechtionscoding** [reply]
Hm. Okay. So what's the logic for finding the data source, it's not declared by the object anywhere per se.

I.e. if I have a helm chart stored in our own artifactory, how does renovate know it's an artifactory data source? The chartUrl? When it makes a call to the source it parses the headers? Host rules?

**Reply by rarkins** [reply]
Look at the `packageFiles with updates` log message, it will include a `datasource` field

**Reply by knechtionscoding** [reply]
Hm. So I'm not seeing that in normal logs. Do I have to set it to debug to get those? I did find one in where someone hadn't put a version, so it was erroring. For that log it just listed datasource as helm. That feels incorrect, or I misunderstood how the datasource should work.

**Reply by rarkins** [reply]
Set `LOG_LEVEL=debug` in env

**Reply by knechtionscoding** [reply]
Ah. If I set debug logs then I can see the releaseTimestamp. Perfect. Thanks
